source/auth/auth.rst

```diff
@@ -683,6 +683,259 @@ mechanism
+
+================= =========
+Host              Region
+================= =========
+sts.amazonaws.com us-east-1
+first.second      second
+first             us-east-1
+================= ========= 
+
+`MongoCredential`_ Properties
+`````````````````````````````
+
+username
+	MAY be specified. The non-sensitive IAM access key.
+
+source
+	MUST be "$external". Defaults to ``$external``.
+
+password
+	MAY be specified. The sensitive IAM secret key.
+
+mechanism
+	MUST be "MONGODB-IAM"
+
+mechanism_properties
+	AWS_SESSION_TOKEN
+		Drivers MUST allow the user to specify an AWS session token for authentication with temporary credentials.
+
+
+Obtaining Credentials
+`````````````````````
+IAM Credentials
+   If a username and password is provided drivers MUST use these for the IAM access key and IAM secret key, respectively. Drivers MUST 
+   use this secret key to generate a signature using HMAC-SHA256. If a username is provided without a password (or vice-versa) drivers 
+   MUST raise an error. An example URI for authentication with MONGODB-IAM using IAM credentials is as follows:
+
+   ``mongodb://<access_key>:<secret_key>@mongodb.example.com/?authMechanism=MONGODB-IAM``
+
+Temporary IAM Credentials
+   If a username and password are not provided drivers MUST query the standard local AWS endpoint for temporary credentials. If 
+   temporary credentials cannot be obtained then drivers MUST fail authentication and raise an error. If an `AWS_SESSION_TOKEN` is
+   provided in addition to a username and password this is considered an `Assume Role <https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html>`_ request. An example URI for authentication with MONGODB-IAM using temporary IAM credentials from an EC2 instance
+   or ECS tasks is as follows:
+
+   ``mongodb://mongodb.example.com/?authMechanism=MONGODB-IAM``
+
+   An example using temporary IAM credentials as an Assume Role request is as follows:
+
+   ``mongodb://<access_key>:<secret_key>@mongodb.example.com/?authMechanism=MONGODB-IAM&authMechanismProperties=AWS_SESSION_TOKEN:<security_token>``
+
+.. note::
+   "session token" and "security token" are used interchangeably here and throughout the AWS documentation.
+
+   EC2 Instance
+      Calling the endpoint ``http://169.254.169.254/latest/meta-data/iam/security-credentials/`` from an EC2 instance 
+      will return the role attached to the EC2 instance in plaintext, if it exist:
+
+      ``role-name``
+
+      ``http://169.254.169.254/latest/meta-data/iam/security-credentials/<role-name>`` will return the following json response: 
+
+      .. code:: javascript
+
+            {
+                "Code": "Success",
+                "LastUpdated" : <date>,
+                "Type": "AWS-HMAC",
+                "AccessKeyId" : <access_key>,
+                "SecretAccessKey": <secret_access_key>,
+                "Token" : <security_token>,
+                "Expiration": <date>
+            }
+
+      .. note::
+
```

Comment:
There's an extra line here hence why Travis is failing: 
```
$ rstcheck -r source/ --ignore-language c,python,java --report error
source/auth/auth.rst:911: (ERROR/3) Content block expected for the "note" directive; none found.
```

Reply:
Done. Also, yes I think it'd be best to add tests in a separate PR since we don't currently have a way to mock the JSON responses for the link-local addresses in evergreen.

Reply:
Okay, can you create a SPEC ticket for adding tests if there isn't one already?